File: dfs_bfs/네트워크.py
from collections import deque
def solution(n, computers):
    answer = 0
    visited = [False] * n
    def bfs(x):
        q = deque([x])
        while q:
            now = q.popleft()
            visited[now] = True
            for i in range(n):
                if computers[now][i] == 1 and not visited[i]:
                    q.append(i)       
    for i in range(n):
        if not visited[i]:
            bfs(i)
            answer += 1
    return answer

File: dfs_bfs/네트워크.java
class Solution {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        
        boolean[] visited = new boolean[n];
        
        for (int i = 0; i < n; i++) {
            if (visited[i]) continue;
            dfs(n, i, computers, visited);
            answer++;
        }
        
        return answer;
    }
    
    public void dfs(int n, int i, int[][] computers, boolean[] visited) {
        visited[i] = true;
        
        for (int j = 0; j < n; j++) {
            if (visited[j]) continue;
            if (computers[i][j] == 1) {
                dfs(n, j, computers, visited);
            }
        }
    }
}
